core/lb.go

```diff
@@ -178,35 +178,37 @@ func (sess *transcoderSession) loop() {
 		select {
 		case <-ctx.Done():
 			// Terminate the session after a period of inactivity
-			glog.V(common.DEBUG).Info("LB: Transcode loop timed out for ", sess.key)
+			clog.V(common.DEBUG).Infof(logCtx, "LB: Transcode loop timed out for key=%s", sess.key)
 			return
 		case params := <-sess.sender:
 			cancel()
 			res, err :=
-				sess.transcoder.Transcode(params.md)
+				sess.transcoder.Transcode(params.ctx, params.md)
```

Comment:
Hum, I think I understood the need but it's quite confusing to have 2 (3?) contexts here. Seems like this context passed here should be `logCtx` instead? Since it's a function that did not receive a ctx before so it's likely only used for logging.

Is there any way of getting rid of the duplication or maybe sticking to only 1 context in the implementation here? Havent read the whole logic so not sure if I follow it, but if we need `params.ctx` for cancellation and `logCtx` for logging, shouldn't `clog.Clone(params.ctx, logCtx)` or sth like that be the only ctx we need here?

Reply:
`Infofe` made for the cases when we don't know if `err` is nil or not. But here we already checked, so no point of using `Infofe`

> Either way came to my mind that Infofe will be logging errors as info, not as errors, right? Sth we'd like to change somehow?


Fixed in 4d312d65ef880fa79dc660cee79d20b7d44945c8

> Also, while at it, mind adding any text here to say something about what is this error?

This concrete error already has verbose description added (like ` errors.Wrapf(err, "Could not get orchestrator orch=%v", orchestratorServer)`)